include/pybind11/eigen.h

```diff
@@ -207,11 +213,56 @@ template <typename Type_> struct EigenProps {
 template <typename props> handle eigen_array_cast(typename props::Type const &src, handle base = handle(), bool writeable = true) {
     constexpr ssize_t elem_size = sizeof(typename props::Scalar);
     array a;
-    if (props::vector)
-        a = array({ src.size() }, { elem_size * src.innerStride() }, src.data(), base);
-    else
-        a = array({ src.rows(), src.cols() }, { elem_size * src.rowStride(), elem_size * src.colStride() },
-                  src.data(), base);
+    using Scalar = typename props::Type::Scalar;
+    if (props::vector) {
```

Comment:
BTW minor formatting nit Could you (a) use `is_pyobject` and (b) "transpose" and reverse this nested `if` statement, just to keep the original code close together?
e.g.
```
if (!is_pyobject) {
   if (props::vector) { ... }
   else { ... }
} else {
   ...  // new code
}
```

Reply:
Done.